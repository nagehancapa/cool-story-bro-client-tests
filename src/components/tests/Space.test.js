import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import Space from "../Space";

describe("<Space >", () => {
  const props = {
    backgroundColor: "#435234",
    color: "#435234",
    title: "My space!",
    description: "some description",
    id: 1,
  };

  test("should render without props", () => {
    const component = renderer.create(<Space />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render with all props", () => {
    const component = renderer.create(<Space {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("A link should render when showLink prop is set to true", () => {
    const component = renderer.create(
      <MemoryRouter>
        <Space {...props} showLink={true} />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
