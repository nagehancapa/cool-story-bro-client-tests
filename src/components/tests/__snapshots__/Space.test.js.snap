// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Space > A link should render when showLink prop is set to true 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": "#435234",
      "color": "#435234",
    }
  }
>
  <h1>
    My space!
  </h1>
  <p>
    some description
  </p>
  <a
    href="/spaces/1"
    onClick={[Function]}
  >
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      Visit space
    </button>
  </a>
</div>
`;

exports[`<Space > should render with all props 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": "#435234",
      "color": "#435234",
    }
  }
>
  <h1>
    My space!
  </h1>
  <p>
    some description
  </p>
</div>
`;

exports[`<Space > should render without props 1`] = `
<div
  className="jumbotron"
  style={
    Object {
      "backgroundColor": undefined,
      "color": undefined,
    }
  }
>
  <h1 />
  <p />
</div>
`;
